src/razor/syntaxes/aspnetcorerazor.tmLanguage.yml

```diff
@@ -420,6 +423,36 @@ repository:
       3: { patterns: [ include: 'source.cs#type' ] }
       4: { name: 'entity.name.variable.property.cs' }
 
+  #>>>>> @rendermode <<<<<
+
+  rendermode-directive:
+    name: 'meta.directive'
+    match: '(@)(rendermode)\s+([^$]+)?'
+    captures:
+      1: { patterns: [ include: '#transition' ] }
+      2: { name: 'keyword.control.razor.directive.rendermode'}
+      3: { patterns: [ include: 'source.cs#type' ] }
+
+  #>>>>> @preservewhitespace <<<<<
+
+  preservewhitespace-directive:
+    name: 'meta.directive'
+    match: '(@)(preservewhitespace)\s+([^$]+)?'
+    captures:
+      1: { patterns: [ include: '#transition' ] }
+      2: { name: 'keyword.control.razor.directive.preservewhitespace'}
+      3: { patterns: [ include: 'source.cs#type' ] }
```

Comment:
`preserveWhitespace` only allows `true` or `false` after it, not a C# type. I think the regex can probably just be updated to `(true|false)` instead of `([^$]+)`, but I don't know what classification name to give. Might need to look in the C# grammar and see what it uses

Reply:
Updating the regex might not be what we want. since TextMate will color the whole line once a match is found. I kept the same regex so that the `@preserveWhitespace` can be colored as soon as the user type a space bar after the keyword, but I updated the last matching pattern for the line to be a `boolean-literal`, so it will change colors if it sees a `true` or a `false`. For visualization please see the updated picture in the PR description. Thanks!

Reply:
Even better! Great work.